java/test/src/main/java/test/Ice/metrics/AllTests.java

```diff
@@ -527,13 +524,25 @@ static MetricsPrx allTests(test.TestHelper helper, CommunicatorObserverI obsv)
       map = toMap(serverMetrics.getMetricsView("View").returnValue.get("Connection"));
       test(map.get("holding").current == 1);
 
-      metrics.ice_getConnection().close(com.zeroc.Ice.ConnectionClose.GracefullyWithWait);
+      // `close` blocks the calling thread until the connection closure is complete.
+      // So, to test the `Closing` state, we call `close` in a new thread, and wait a little.
+      var closureThread = new Thread(() -> metrics.ice_getConnection().close());
+      closureThread.start();
+      try {
+        Thread.sleep(50);
+      } catch (InterruptedException ex) {
+      }
 
       map = toMap(clientMetrics.getMetricsView("View").returnValue.get("Connection"));
       test(map.get("closing").current == 1);
       map = toMap(serverMetrics.getMetricsView("View").returnValue.get("Connection"));
       test(map.get("holding").current == 1);
 
+      try {
+        // Wait for the connection closure thread to finish, so we're in a known state again.
+        closureThread.join();
+      } catch (InterruptedException ex) {
+      }
```

Comment:
How can closure complete before we resume from hold?

Reply:
Yes, you're correct, I had them in the incorrect order...
The test still worked because we hit the CloseTimeout, but it did hang the test for 10s for no reason. Not great!

Fixed.